app/assets/javascripts/modules/sticky-element-container.js

```diff
@@ -0,0 +1,136 @@
+/*
+  This module will cause a child in the target element to:
+  - hide when the top of the target element is visible;
+  - stick to the bottom of the window while the parent element is in view;
+  - stick to the bottom of the target when the user scrolls past the bottom.
+
+  Use 'data-module="sticky-element-container"' to instantiate, and add
+  `[data-sticky-element]` to the child you want to position.
+*/
+(function (Modules, root) {
+  'use strict';
+
+  var $ = root.$;
+  var $window = $(root);
+
+  Modules.StickyElementContainer = function () {
+    var self = this;
+
+    self._getWindowDimensions = function _getWindowDimensions () {
+      return {
+        height: $window.height(),
+        width: $window.width()
+      };
+    };
+
+    self._getWindowPositions = function _getWindowPositions () {
+      return {
+        scrollTop: $window.scrollTop()
+      };
+    };
+
+    self.start = function start ($el) {
+      var $element = $el.find('[data-sticky-element]');
```

Comment:
This might be clearer if `$element` was `$stickyElement`. I confused it with `$el` that gets passed in to `start()`.